core/sdk-core/src/test/java/software/amazon/awssdk/core/client/config/ClientOverrideConfigurationTest.java

```diff
@@ -278,4 +278,16 @@ public void executionAttributes_listPreviouslyAdded_appendedToList() {
         ClientOverrideConfiguration overrideConfig = builder.build();
         assertThat(overrideConfig.executionAttributes().getAttribute(addedAttribute)).isEqualTo(addedValue);
     }
+
+    @Test
+    void builder_headersWithEmptyAdditionalHeaders_shouldReturnEmptyMap_shouldNotThrowNullPointerException() {
+        ClientOverrideConfiguration.Builder builder = ClientOverrideConfiguration.builder();
+        Map<String, List<String>> headers = builder.headers();
+        assertThat(headers).isEmpty(); // <--- should not throw NPE
```

Comment:
```suggestion
        assertThat(headers).isNotNull();
```
Feels better to add an explicit assertion of what you expect instead of leaving it to a comment.

Reply:
The thing is, it's not the `headers` returned by `.headers()` that is null and causing a problem, but the map that this map returns. The NPE is thrown when the top level Map interacts with its delegate.

Reply:
Yeah, that's why the suggestion above.

> Can we also add a `Validate.paramNotNull(⋯)` [here](https://github.com/aws/aws-sdk-java-v2/blob/e5570f717085d451782c0c1ea0006f8ac35a15b3/utils/src/main/java/software/amazon/awssdk/utils/UnmodifiableMapOfLists.java#L43)?